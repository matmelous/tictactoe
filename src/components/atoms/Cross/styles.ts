import styled from "styled-components";

export const CrossStyled = styled.div`
  width: 100px;
  height: 100px;
  transform: scale(0.7);

  .neon {
    width: 4px;
    border-radius: 10px;
    box-shadow: inset 0px 0px 2px #000000, inset 0px 0px 1px #000000,
      0px 0px 10px #31878d, 0px 0px 5px #31878d;
    height: 100px;
    background-color: #eefcfc;
  }
`;

export const Arm1 = styled.div`
  transform: translate(0px, 50px) rotate(45deg);
  display: flex;
  height: 4px;
  width: 100px;
  justify-content: space-between;
  align-items: space-between;

  .neon {
    height: 4px;
    width: 47px;
  }
`;
export const Arm2 = styled.div`
  transform: rotate(45deg) translate(32px, -35px);
`;
