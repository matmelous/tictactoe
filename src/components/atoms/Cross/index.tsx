import React from "react";
import { Arm1, Arm2, CrossStyled } from "./styles";

const Cross = () => {
  return (
    <CrossStyled>
      <Arm1>
        <div className="neon"></div>
        <div className="neon"></div>
      </Arm1>
      <Arm2 className="neon"></Arm2>
    </CrossStyled>
  );
};

export default Cross;
