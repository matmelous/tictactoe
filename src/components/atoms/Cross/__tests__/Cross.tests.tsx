import React from "react";
import { render, cleanup } from "@testing-library/react";
import Cross from "../index";

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<Cross />);
  expect(asFragment()).toMatchSnapshot();
});