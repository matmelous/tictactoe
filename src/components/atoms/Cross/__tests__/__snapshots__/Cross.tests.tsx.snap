// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL ehIecs"
    role="cross"
  >
    <div
      class="sc-gsDKAQ cAkuNE"
    >
      <div
        class="neon"
      />
      <div
        class="neon"
      />
    </div>
    <div
      class="sc-dkPtRN hzjwCj neon"
    />
  </div>
</DocumentFragment>
`;
